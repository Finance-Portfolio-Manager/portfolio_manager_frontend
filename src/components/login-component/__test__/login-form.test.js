import React from 'react';
import ReactDOM from 'react-dom';
import LoginForm from './../login-form';

import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";

// import renderer from "@testing-library/react";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("renders without crashing", ()=>{
    const div = document.createElement("div");
    ReactDOM.render(<LoginForm></LoginForm>, div)
})

it("renders login correctly", ()=>{
    const div = document.createElement("div");
    const { getByTestId } = render(<LoginForm></LoginForm>, div);
    expect(getByTestId("login-form")).toHaveTextContent("Portfolio ManagerUsername:Password:Login");
})

it("matches snapshot", ()=>{
    const tree = renderer.create(<LoginForm></LoginForm>).toJSON();
    expect(tree).toMatchSnapshot();
})