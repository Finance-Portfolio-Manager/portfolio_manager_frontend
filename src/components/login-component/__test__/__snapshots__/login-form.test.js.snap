// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="App"
  data-testid="login-form"
>
  <form
    className=""
  >
    <div
      className="aroundLogo"
    >
      <img
        alt="revature logo"
        className="logo"
        src="revlogo.jpg"
      />
    </div>
    <div
      className="form-inner"
    >
      <h4>
        Portfolio Manager
      </h4>
      <div
        className="form-group"
      >
        <label
          className="form-label"
          htmlFor="formBasicUsername"
        >
          Username:
        </label>
        <input
          className="loginInput form-control"
          id="formBasicUsername"
          name="username"
          placeholder="Enter your username"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="form-label"
          htmlFor="formBasicPassword"
        >
          Password:
        </label>
        <input
          className="loginInput form-control"
          id="formBasicPassword"
          name="password"
          placeholder="Enter your password"
          type="password"
        />
      </div>
      <button
        className="submitButton mt-2"
        type="submit"
      >
        Login
      </button>
    </div>
  </form>
</div>
`;
